src/acl_hal_mmd.cpp

```diff
@@ -965,22 +996,40 @@ void l_get_physical_devices(acl_mmd_dispatch_t *mmd_dispatch,
   mmd_dispatch->aocl_mmd_get_offline_info(AOCL_MMD_VERSION, sizeof(buf), buf,
                                           NULL);
   buf[sizeof(buf) - 1] = 0;
-  mmd_dispatch->mmd_version = atof(buf);
+  mmd_dispatch->mmd_version = l_parse_mmd_version_str(std::string(buf));
+  if (mmd_dispatch->mmd_version < 0) {
+    printf("  Invalid MMD version:     %s\n", buf);
+    printf("Contact the board package support vendor for resolution.\n");
```

Comment:
@aktole Could you check if the printed message here is okay?

Reply:
Yes, looks good. This condition would probably never be true, unless BSP vendor sets the MMD version to something less than 0 accidently right?